pkg/apis/kops/validation/instancegroup.go

```diff
@@ -0,0 +1,84 @@
+package validation
+
+import (
+	"fmt"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/kops/pkg/apis/kops"
+	"k8s.io/kops/pkg/apis/kops/util"
+)
+
+func ValidateInstanceGroup(g *kops.InstanceGroup) error {
+	if g.ObjectMeta.Name == "" {
+		return field.Required(field.NewPath("Name"), "")
+	}
+
+	if g.Spec.Role == "" {
+		return field.Required(field.NewPath("Role"), "Role must be set")
+	}
+
+	switch g.Spec.Role {
```

Comment:
This switch doesn't seem to be doing anything (it should always hit default).

Just in place for future logic? Or..?